cmd/kubeadm/app/preflight/checks.go

```diff
@@ -831,8 +833,11 @@ func (ImagePullCheck) Name() string {
 // Check pulls images required by kubeadm. This is a mutating check
 func (ipc ImagePullCheck) Check() (warnings, errorList []error) {
 	for _, image := range ipc.imageList {
+		if ipc.NodeRegistration.ImagePullPolicy == string(v1.PullNever) {
```

Comment:
- the string casts are not needed if ImagePullPolicy is of type v1.PullPolicy
- you should refactor this code to use a switch because it makes it easier to read:

```
switch ipc.imagePullPolicy {
case ...
}
```